### PR TITLE
Compile shared sources from folder, not package

### PR DESCRIPTION
**Pull request by natemcmaster**
Looking ahead to the MSBuild future. Remove Microsoft.Extensions.FileProviders.Sources package and compile from shared folder.
AFAIK no other aspnet repos use this package show shouldn't need to keep publishing it on our CI feeds.

cref https://github.com/aspnet/Coherence-Signed/issues/379

cc @pranavkm @Eilon

### CODE REVIEW

File: src/Microsoft.Extensions.FileProviders.Abstractions/project.json [hunk 1]
```diff
@@ -3,7 +3,8 @@
   "buildOptions": {
     "warningsAsErrors": true,
     "keyFile": "../../tools/Key.snk",
-    "xmlDoc": true
+    "xmlDoc": true,
+    "compile": "../Shared/**/*.cs"
```
Comment (pranavkm) [inline src/Microsoft.Extensions.FileProviders.Abstractions/project.json]: lower case "shared"? (Since we have `src`, `tools`, `test` etc)

**Comment by natemcmaster** [comment]
Ok, will wait for #250 and then remove the dead code.

Re: `EmptyDisposable`. Seems like a good API for .NET to add or to move into aspnet/Common. It's not filesystem specific.

File: test/Microsoft.Extensions.FileProviders.Composite.Tests/project.json [hunk 2]
```diff
@@ -1,9 +1,14 @@
...
+    "warningsAsErrors": true,
+    "keyFile": "../../tools/Key.snk",
+    "compile": "../../shared/**/*.cs"
+  },
   "dependencies": {
     "dotnet-test-xunit": "2.2.0-*",
     "Microsoft.AspNetCore.Testing": "1.2.0-*",
     "Microsoft.Extensions.FileProviders.Composite": "1.2.0-*",
-    "Microsoft.Extensions.FileProviders.Sources": "1.2.0-*",
+    "Moq": "4.6.38-alpha",
```
Comment (JunTaoLuo) [inline test/Microsoft.Extensions.FileProviders.Composite.Tests/project.json]: Is there a feature we are using from this alpha version? I see else where in our repos we're using another version: https://github.com/aspnet/BasicMiddleware/blob/ea0042c685ba2228d0d4b85df41fa45a0b360f00/test/Microsoft.AspNetCore.ResponseCompression.Tests/project.json#L15

Comment (natemcmaster) [inline test/Microsoft.Extensions.FileProviders.Composite.Tests/project.json]: 4.6.36 is also an alpha. We aren't uniform on this one though. Other repos use the .38 patch.

**Comment by natemcmaster** [comment]
Yes, Jass is working on removing some of this duplication in #250. This PR is mostly about getting off the "shared" sources project.json model since that package layout will not work in vs2017.

**Pull request merged**